### ISSUE TITLE
切换至最新版本代码后，相同训练参数，grad_norm与kl为0

### ISSUE THREAD

**Issue by vincezengqiang** [modelscope/ms-swift#7229]
swift rlhf \
    --rlhf_type grpo \
    --model_type qwen3 \
    --importance_sampling_level sequence \
    --template qwen3 \
    --use_vllm true \
    --vllm_mode colocate \
    --beta 0.001 \
    --epsilon 3e-4 \
    --epsilon_high 4e-4 \
    --steps_per_generation 4 \
    --vllm_gpu_memory_utilization 0.65 \
    --tensor_parallel_size 4 \
    --num_generations 4 \
    --vllm_pipeline_parallel_size 2 \
    --vllm_data_parallel_size 8 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-6 \
    --gradient_accumulation_steps 4 \
    --save_only_model true \
    --num_train_epochs 2 \
    --save_total_limit 6 \
    --save_strategy epoch \
    --logging_steps 5 \
    --max_length 25600 \
    --output_dir 'xxx' \
    --model 'xxx' \
    --external_plugins /checkpoint/binary/train_package/examples/train/grpo/plugin/plugin.py \
    --reward_funcs AccORM SoftOverlong_fix \
    --reward_weights 1 0.2 \
    --soft_max_length 350 \
    --soft_cache_length 100 \
    --soft_min_length 80 \
    --soft_short_penalty 50 \
    --train_type full \
    --torch_dtype bfloat16 \
    --dataset 'xxx' \
    --max_completion_length 512 \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 8 \
    --dataset_num_proc 8 \
    --temperature 0.8 \
    --log_completions true \
    --log_entropy true \
    --deepspeed zero3 \
    --report_to tensorboard \
    --dataset_shuffle false \
    --response_prefix '<think>\n\n</think>\n\n' \
    --seed 500 \
    --data_seed 500 \
    --attn_impl flash_attn
使用这套参数训练时，训练日志如下：
{'loss': 0.00034841, 'grad_norm': 0.20733885, 'learning_rate': 0.0, 'reward': 1.22858746, 'reward_std': 0.16965312, 'frac_reward_zero_std': 0.0, 'rewards/AccORM/mean': 0.7359375, 'rewards/AccORM/std': 0.4212365, 'rewards/SoftOverlong_fix/mean': -0.01721875, 'rewards/SoftOverlong_fix/std': 0.09705251, 'rewards/ConsistencyReward/mean': 0.9921875, 'rewards/ConsistencyReward/std': 0.07793439, 'completions/mean_length': 142.6359375, 'completions/min_length': 84.0, 'completions/max_length': 320.6, 'completions/clipped_ratio': 0.0015625, 'entropy/mean': 0.14541016, 'entropy/max': 0.28208008, 'entropy/min': 0.0645752, 'entropy/threshold': 0.15602299, 'kl': 0.33107089, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 10.0, 'global_step/max_steps': '580/580', 'percentage': '100.00%', 'elapsed_time': '11h 17m 24s', 'remaining_time': '0s', 'memory(GiB)': 73.84, 'train_speed(iter/s)': 0.01427}

**Comment by vincezengqiang** [comment]
grad_norm与kl正常

切换到最新代码，使用相同脚本训练时，日志：
{'loss': -1e-08, 'grad_norm': 0.0, 'learning_rate': 0.0, 'completions/mean_length': 145.9796875, 'completions/min_length': 70.6, 'completions/max_length': 334.0, 'completions/clipped_ratio': 0.0, 'reward': 0.94189374, 'reward_std': 0.3534429, 'frac_reward_zero_std': 0.29375, 'rewards/AccORM/mean': 0.45625, 'rewards/AccORM/std': 0.49048966, 'rewards/SoftOverlong_fix/mean': -0.01240625, 'rewards/SoftOverlong_fix/std': 0.09140642, 'rewards/ConsistencyReward/mean': 0.97625, 'rewards/ConsistencyReward/std': 0.14700245, 'entropy/mean': 0.28696289, 'entropy/max': 0.59150391, 'entropy/min': 0.12155762, 'kl': 0.0, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 10.0, 'global_step/max_steps': '580/580', 'percentage': '100.00%', 'elapsed_time': '14h 11m 16s', 'remaining_time': '0s', 'memory(GiB)': 95.13, 'train_speed(iter/s)': 0.011356}

grad_norm与kl始终为0，请问是为什么？感谢指教🙏

**Comment by hjh0119** [comment]
run normally on my end 

```
CUDA_VISIBLE_DEVICES=0,1,2,3 \
NPROC_PER_NODE=4 \
swift rlhf \
    --rlhf_type grpo \
    --model Qwen/Qwen3-4B \
    --model_type qwen3 \
    --importance_sampling_level sequence \
    --template qwen3 \
    --use_vllm true \
    --vllm_mode colocate \
    --sleep_level 1 \
    --beta 0.001 \
    --epsilon 3e-4 \
    --epsilon_high 4e-4 \
    --steps_per_generation 4 \
    --vllm_gpu_memory_utilization 0.3 \
    --vllm_max_model_len 16384 \
    --num_generations 4 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-6 \
    --gradient_accumulation_steps 4 \
    --save_only_model true \
    --num_train_epochs 2 \
    --save_total_limit 6 \
    --save_strategy epoch \
    --logging_steps 5 \
    --max_length 8192 \
    --reward_funcs accuracy format \
    --train_type full \
    --torch_dtype bfloat16 \
    --dataset 'AI-MO/NuminaMath-TIR#1000' \
    --max_completion_length 8192 \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 8 \
    --dataset_num_proc 8 \
    --temperature 0.8 \
    --log_completions true \
    --log_entropy true \
    --deepspeed zero3 \
    --report_to tensorboard swanlab \
    --dataset_shuffle false \
    --response_prefix '<think>\n\n</think>\n\n' \
    --seed 500 \
    --data_seed 500 \
    --attn_impl flash_attn
```




BTW, it looks like you've modified the code — the following parameters are invalid/illegal.

```
    --tensor_parallel_size 4 \
    --vllm_pipeline_parallel_size 2 \
    --vllm_data_parallel_size 8 \
    --soft_min_length 80 \
    --soft_short_penalty 50 \
```
